cmd/lambda/leases/create.go

```diff
@@ -2,203 +2,95 @@ package main
 	user := r.Context().Value(api.User{}).(*api.User)
-	err = user.Authorize(principalID)
+	err = user.Authorize(*newLease.PrincipalID)
 	if err != nil {
 		api.WriteAPIErrorResponse(w, err)
 		return
 	}
 
-	log.Printf("Creating lease for Principal %s", principalID)
-
-	// Fail if the Principal already has an active lease
-	principalLeases, err := dao.FindLeasesByPrincipal(requestBody.PrincipalID)
-	if err != nil {
-		log.Printf("Failed to list leases for principal %s: %s", requestBody.PrincipalID, err)
-		response.WriteServerErrorWithResponse(w, "Internal server error")
-		return
-	}
-
-	for _, lease := range principalLeases {
-		if lease.LeaseStatus == db.Active {
-			msg := fmt.Sprintf("Principal already has an active lease for account %s", lease.AccountID)
-			response.WriteConflictError(w, msg)
-			return
-		}
+	// Get the First available Ready Account
+	query := &account.Account{
+		Status: account.StatusReady.StatusPtr(),
 	}
 
-	// Get the First Ready Account
-	// Exit if there's an error or no ready accounts
-	account, err := dao.GetReadyAccount()
+	accounts, err := Services.AccountService().List(query)
 	if err != nil {
-		log.Printf("Failed to Check Ready Accounts: %s", err)
-		response.WriteServerErrorWithResponse(
-			w,
-			fmt.Sprintf("Failed to find a Ready Account: %s", err),
-		)
-		return
-	} else if account == nil {
-		errStr := "No Available accounts at this moment"
-		log.Println(errStr)
-		response.WriteServiceUnavailableError(w, errStr)
+		api.WriteAPIErrorResponse(w, err)
 		return
 	}
-	log.Printf("Principal %s will be Leased to Account: %s\n", principalID,
-		account.ID)
-
-	// Create/Update lease record
-	now := time.Now()
-	lease, err := dao.UpsertLease(db.Lease{
-		AccountID:                account.ID,
-		PrincipalID:              requestBody.PrincipalID,
-		ID:                       uuid.New().String(),
-		LeaseStatus:              db.Active,
-		LeaseStatusReason:        db.LeaseActive,
-		BudgetAmount:             requestBody.BudgetAmount,
-		BudgetCurrency:           requestBody.BudgetCurrency,
-		BudgetNotificationEmails: requestBody.BudgetNotificationEmails,
-		CreatedOn:                now.Unix(),
-		LastModifiedOn:           now.Unix(),
-		LeaseStatusModifiedOn:    now.Unix(),
-		ExpiresOn:                requestBody.ExpiresOn,
-		Metadata:                 requestBody.Metadata,
-	})
-	if err != nil {
-		log.Printf("Failed to create lease DB record for %s @ %s: %s",
-			requestBody.PrincipalID, account.ID, err)
-		response.WriteServerError(w)
+	if accounts != nil && len(*accounts) == 0 {
+		api.WriteAPIErrorResponse(w,
+			errors.NewInternalServer("No Available accounts at this moment", nil))
 		return
 	}
+	availableAccount := (*accounts)[0]
 
-	// Mark the accOunt as Status=Leased
-	_, err = dao.TransitionAccountStatus(account.ID, db.Ready, db.Leased)
+	// Get user principal's current spend
+	usageStartTime := getBeginningOfCurrentBillingPeriod(Settings.PrincipalBudgetPeriod)
```

Comment:
BTW, I see that it may have been _me_ that put this method here. What can I say? Sometimes it's easier to see things you'd do differently after you take a step back from the situation...

Reply:
We can centralize this, as part of usage2.0.